File: GXpert/GXpert.Web/Modules/Masters/AcademicPerformance/AcademicPerformanceColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.AcademicPerformance")]
[BasedOnRow(typeof(AcademicPerformanceRow), CheckNames = true)]
public class AcademicPerformanceColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string StudentPrn { get; set; }
    [QuickFilter]
    public string CourseTitle { get; set; }
    [QuickFilter]
    public string ClassTitle { get; set; }
    [QuickFilter]
    public string SemesterTitle { get; set; }
    public float MarksObtained { get; set; }
    public float OutOfMarks { get; set; }
    [EditLink]
    public string Remarks { get; set; }
    public string AcademicYearName { get; set; }
    
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/AcademicPerformanceColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AcademicPerformanceRow } from "./AcademicPerformanceRow";

export interface AcademicPerformanceColumns {
    Id: Column<AcademicPerformanceRow>;
    StudentPrn: Column<AcademicPerformanceRow>;
    CourseTitle: Column<AcademicPerformanceRow>;
    ClassTitle: Column<AcademicPerformanceRow>;
    SemesterTitle: Column<AcademicPerformanceRow>;
    MarksObtained: Column<AcademicPerformanceRow>;
    OutOfMarks: Column<AcademicPerformanceRow>;
    Remarks: Column<AcademicPerformanceRow>;
    AcademicYearName: Column<AcademicPerformanceRow>;
}

export class AcademicPerformanceColumns extends ColumnsBase<AcademicPerformanceRow> {
    static readonly columnsKey = 'Masters.AcademicPerformance';
    static readonly Fields = fieldsProxy<AcademicPerformanceColumns>();
}